docs/storage/volume_migration.md

```diff
@@ -0,0 +1,219 @@
+
+The volume migration depends on the `VolumeMigration` feature gate. To enable the feature gate, add the following to the KubeVirt CR:
+```yaml
+apiVersion: kubevirt.io/v1
+kind: KubeVirt
+spec:
+  configuration:
+    developerConfiguration:
+      featureGates:
+        - VMLiveUpdateFeatures
+        - VolumesUpdateStrategy
+        - VolumeMigration
+  workloadUpdateStrategy:
+    workloadUpdateMethods:
+    - LiveMigrate
+```
+
+## Datavolume template
+
+Updating a datavolume that is referenced by a datavolume template requires special caution. The volumes section must include a reference to the name of the datavolume template. This means that the datavolume templates must either be entirely deleted or updated as well.
+
+Example of updating the datavolume for the original VM in the first example:
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: DataVolume
+metadata:
+  name: dst-dv
+spec:
+  source:
+      blank: {}
+  pvc:
+    accessModes:
+      - ReadWriteOnce
+    resources:
+      requests:
+        storage: 2Gi
+    storageClassName: local
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  labels:
+    kubevirt.io/vm: vm-dv
+  name: vm-dv
+spec:
+  updateVolumesStrategy: Migration
+  dataVolumeTemplates:
+  - metadata:
+      name: dst-dv
+    spec:
+      pvc:
+        accessModes:
+        - ReadWriteOnce
+        volumeMode: Filesystem
+        resources:
+          requests:
+            storage: 2Gi
+        storageClassName: local
+      source:
+        registry:
+          url: docker://registry:5000/kubevirt/alpine-container-disk-demo:devel
+  running: true
+  template:
+    metadata:
+      labels:
+        kubevirt.io/vm: vm-dv
+    spec:
+      domain:
+        devices:
+          disks:
+          - disk:
+              bus: virtio
+            name: datavolumedisk1
+          interfaces:
+          - masquerade: {}
+            name: default
+        resources:
+          requests:
+            memory: 128Mi
+      terminationGracePeriodSeconds: 0
+      volumes:
+      - dataVolume:
+          name: dst-dv
+        name: datavolumedisk1
+      networks:
+      - name: default
+        pod: {}
+```
+
+## Limitations
+
+Only certain types of disks and volumes are supported to be migrated. For an invalid type of volume the RestartRequired condition is set and volumes will be replaced upon VM restart.
+Currently, the volume migration is supported between PersistentVolumeClaims and Datavolumes.
+Additionally, volume migration is forbidden if the volumes is:
+* shareable, since it cannot guarantee the data consistency with multiple writers
+* hotpluggable, this case isn't currently supported
+* filesystem, since virtiofs doesn't curretly support live-migration
```

Comment:
So the source or the destination volume can't be filesystem?

Reply:
Storage migration is currently supported for filesystem types

Reply:
Oh, okay, so filesystem is not forbidden, and this line can be removed from the list? 

> Additionally, volume migration is forbidden if the volumes is:
> ...
> * filesystem, since virtiofs doesn't curretly support live-migration

Reply:
Sorry, no it isn't support, but here we mean the disk type filesystem no the filesystem PVC. Virtiofs doesn't support migration yet